staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -0,0 +1,308 @@
+	ClusterSubresource(resource schema.GroupVersionResource, subresource string) DynamicResourceInterface
+	// Deprecated, this isn't how we want to do it
+	NamespacedSubresource(resource schema.GroupVersionResource, subresource, namespace string) DynamicResourceInterface
+}
+
+type DynamicResourceInterface interface {
+	Create(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+	Update(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+	UpdateStatus(obj *unstructured.Unstructured) (*unstructured.Unstructured, error)
+	Delete(name string, options *metav1.DeleteOptions) error
+	DeleteCollection(options *metav1.DeleteOptions, listOptions metav1.ListOptions) error
+	Get(name string, options metav1.GetOptions) (*unstructured.Unstructured, error)
+	List(opts metav1.ListOptions) (*unstructured.UnstructuredList, error)
+	Watch(opts metav1.ListOptions) (watch.Interface, error)
+	Patch(name string, pt types.PatchType, data []byte, subresources ...string) (*unstructured.Unstructured, error)
+}
+
+type DynamicClient struct {
+	client *rest.RESTClient
+}
+
+var _ DynamicInterface = &DynamicClient{}
+
+func NewForConfig(inConfig *rest.Config) (DynamicInterface, error) {
+	config := rest.CopyConfig(inConfig)
+	// for serializing the options
+	config.GroupVersion = &schema.GroupVersion{}
+	config.APIPath = "/if-you-see-this-search-for-the-break"
+	config.AcceptContentTypes = "application/json"
+	config.ContentType = "application/json"
+	config.NegotiatedSerializer = unstructuredscheme.DefaultUnstructuredNegotiatedSerializer()
+	if config.UserAgent == "" {
+		config.UserAgent = rest.DefaultKubernetesUserAgent()
+	}
+
+	restClient, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DynamicClient{client: restClient}, nil
+}
+
+type DynamicResourceClient struct {
+	client      *DynamicClient
+	namespace   string
+	resource    schema.GroupVersionResource
+	subresource string
+}
+
+func (c *DynamicClient) ClusterResource(resource schema.GroupVersionResource) DynamicResourceInterface {
+	return &DynamicResourceClient{client: c, resource: resource}
+}
+func (c *DynamicClient) NamespacedResource(resource schema.GroupVersionResource, namespace string) DynamicResourceInterface {
+	return &DynamicResourceClient{client: c, resource: resource, namespace: namespace}
+}
+
+func (c *DynamicClient) ClusterSubresource(resource schema.GroupVersionResource, subresource string) DynamicResourceInterface {
+	return &DynamicResourceClient{client: c, resource: resource, subresource: subresource}
+}
+func (c *DynamicClient) NamespacedSubresource(resource schema.GroupVersionResource, subresource, namespace string) DynamicResourceInterface {
+	return &DynamicResourceClient{client: c, resource: resource, namespace: namespace, subresource: subresource}
+}
+
+func (c *DynamicResourceClient) Create(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	if len(c.subresource) > 0 {
```

Comment:
you can post to subresources (e.g. https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/storage.go#L137-L149)

the name and subresource have to be specified together

Reply:
Follow-up.  The subresource here only exists to ease the old dynamic client out